docs/user_guide.md

```diff
@@ -0,0 +1,37 @@
+# Using S2I images
+
+## Invoking scripts embedded into an image
+
+When there is a requirement for **adding custom logic** to [S2I scripts](builder_image.md#s2i-scripts), it is possible to execute standartd scripts inside of custom scripts from user’s repository. Note that it only works when you have to add logic (for example, to generate some static assets after executing `assemble` script) and not for the case of modifying this logic. In latter case you have to replace standard scripts.
```

Comment:
this first sentence is pretty confusing.  
1) "when there is a requirement for adding custom logic to s2i scripts"?  I think what you mean here is basically "Most S2I builder images include assemble/run scripts in the image.  If there is a need to augment the steps being performed by those scripts, there are two options:  1) replace the script entirely by including your own scripts in your source repo or 2) include a new script in your source repo that performs the additional steps before/after invoking the built in scripts (note that for run script, only invoking before is valid).  Option 2 is only applicable when logic is being added, not changed, and when the logic can be invoked before or after the existing script logic.  To change existing logic or inject additional actions in the middle of the existing logic, the entire script must be replaced/overridden."

something like that, anyway. please don't take it verbatim, i'm typing off the top of my head here :)